pkg/controllers/streaming/processor_controller.go

```diff
@@ -18,36 +18,225 @@ package streaming
 )
 
 // ProcessorReconciler reconciles a Processor object
 type ProcessorReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	Tracker tracker.Tracker
 }
 
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 
 func (r *ProcessorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("processor", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("processor", req.NamespacedName)
 
-	// your logic here
+	var original streamingv1alpha1.Processor
+	if err := r.Client.Get(ctx, req.NamespacedName, &original); err != nil {
+		return ctrl.Result{}, ignoreNotFound(err)
+	}
 
-	return ctrl.Result{}, nil
+	// Don't modify the informers copy
+	processor := original.DeepCopy()
+
+	// Reconcile this copy of the processor and then write back any status
+	// updates regardless of whether the reconciliation errored out.
+	result, err := r.reconcile(ctx, log, processor)
+
+	// check if status has changed before updating, unless requeued
+	if !result.Requeue && !equality.Semantic.DeepEqual(original.Status, processor.Status) {
+		if updateErr := r.Status().Update(ctx, processor); updateErr != nil {
+			log.Error(updateErr, "unable to update Processor status", "processor", processor)
+			return ctrl.Result{Requeue: true}, updateErr
+		}
+	}
+	return result, err
 }
 
 func (r *ProcessorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueTrackedResources := &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+			requests := []reconcile.Request{}
+			for _, item := range r.Tracker.Lookup(a.Object.(metav1.ObjectMetaAccessor)) {
+				requests = append(requests, reconcile.Request{NamespacedName: item})
+			}
+			return requests
+		}),
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&streamingv1alpha1.Processor{}).
+		Owns(&appsv1.Deployment{}).
+		Watches(&source.Kind{Type: &buildv1alpha1.Function{}}, enqueueTrackedResources).
 		Complete(r)
 }
+
+func (r *ProcessorReconciler) reconcile(ctx context.Context, logger logr.Logger, processor *streamingv1alpha1.Processor) (ctrl.Result, error) {
+	if processor.GetDeletionTimestamp() != nil {
+		return ctrl.Result{}, nil
+	}
+
+	// We may be reading a version of the object that was stored at an older version
+	// and may not have had all of the assumed defaults specified.  This won't result
+	// in this getting written back to the API Server, but lets downstream logic make
+	// assumptions about defaulting.
+	processor.SetDefaults(ctx)
+
+	processor.Status.InitializeConditions()
+
+	// resolve function
```

Comment:
pull into a separate method, similar to https://github.com/projectriff/system/blob/87f386572f252b7a04852cdf97f50fd6188b4547/pkg/controllers/core/deployer_controller.go#L127